### ISSUE TITLE
Different resolutions have a great impact on performance improvement

### ISSUE THREAD

**Issue by radish0926** [HaozheLiu-ST/T-GATE#7]
When the image generated by sdxl and sd1.5 is 515*512, tgate can achieve a nearly 50% improvement. However, when I increase the resolution to 1024*1024, the performance improvement is only about 10%.

**Comment by HaozheLiu-ST** [comment]
Hi, thanks for your attention to our work. 

We agree that our method's improvement is related to the input size. 

However, based on our empirical study, we believe that our method works better when increasing the input size rather than decreasing it. 

I list a simple base code for testing:
```
from tgate import TgateSDXLLoader
from diffusers import StableDiffusionXLPipeline
from diffusers import DPMSolverMultistepScheduler
import torch
import time

gate_step = 5
inference_num_per_image = 20
resolution=1024
prompt = "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k"
def initial():
    pipeline_text2image = StableDiffusionXLPipeline.from_pretrained(
        "stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16, variant="fp16", use_safetensors=True
    )
    pipeline_text2image = pipeline_text2image.to("cuda")
    return pipeline_text2image



pipeline_text2image = initial()
generator = torch.manual_seed('10086')
pipeline_text2image.scheduler = DPMSolverMultistepScheduler.from_config(pipeline_text2image.scheduler.config)
for i in range(5):
    image = pipeline_text2image(prompt=prompt, num_inference_steps = inference_num_per_image, generator=generator,height=resolution,width=resolution).images[0]
s_time = time.time()
image = pipeline_text2image(prompt=prompt, num_inference_steps = inference_num_per_image, generator=generator,height=resolution,width=resolution).images[0]
e_time = time.time()
interval = e_time-s_time
print('%.3f second'%interval)
# Results: 5.774s

pipeline_text2image = initial()
pipeline_text2image = TgateSDXLLoader(pipeline_text2image, gate_step=gate_step)
generator = torch.manual_seed('10086')
pipeline_text2image.scheduler = DPMSolverMultistepScheduler.from_config(pipeline_text2image.scheduler.config)
for i in range(5):
    image = pipeline_text2image.tgate(prompt=prompt, num_inference_steps = inference_num_per_image, gate_step=gate_step, generator=generator,height=resolution,width=resolution).images[0]
s_time = time.time()
image = pipeline_text2image.tgate(prompt=prompt, num_inference_steps = inference_num_per_image, gate_step=gate_step, generator=generator,height=resolution,width=resolution).images[0]
e_time = time.time()
interval = e_time-s_time
print('%.3f second'%interval)
# Results: 3.899 second
```

Tgate can improve the baseline from 5.774s to 3.899s on a platform of one V100 GPU card (32.47% improvement). 

When the resolution is set to 512, the improvement is from 1.705s to 1.359s on the same platform (25.46% improvement). 

The latency is highly related to the computation platform, batch size, CUDA version, and pytorch version. We encourage you to use MACs as the primary metric for evaluation, which is more objective. 

A potential reason for only a 10% improvement is missing the warm-up stage. 
```
for i in range(5):
    image = pipeline_text2image.tgate(prompt=prompt, num_inference_steps = inference_num_per_image, gate_step=gate_step, generator=generator,height=resolution,width=resolution).images[0]
```

**Comment by HaozheLiu-ST** [comment]
Hi, please let me know if you still have any issues. If you find this project useful for you, please consider giving it a star. Thanks!
